server/controllers/bookmarkController.js

```diff
@@ -0,0 +1,95 @@
+import sequelize from 'sequelize';
+import models from '../db/models';
+import utils from '../helpers/Utilities';
+
+const { Op } = sequelize;
+
+/**
+ * @Module BookmarkController
+ * @description Controlls all activities related to bookmarks
+ */
+class BookmarkController {
+  /**
+   * @description Bookmarks an article for reading later
+   * @param {Object} req - request object
+   * @param {Object} res - response object
+   * @returns {Object} Success Message
+   */
+  static async bookmarkArticle(req, res) {
+    const userId = req.user.id;
+    const { articleId } = req.params;
+    const article = await models.Article.findByPk(articleId);
+    if (!article) return utils.errorStat(res, 404, 'Article not found');
+    const bookmarked = await models.Bookmark.findOne({
+      where: {
+        [Op.or]: [{ articleId }]
+      }
+    });
+    if (bookmarked) return utils.errorStat(res, 409, 'Aticle already bookmarked');
+    const addBookmark = await models.Bookmark.findOrCreate({
+      where: { articleId, userId },
+      defaults: { articleId, userId }
+    });
+    return utils.successStat(res, 201, 'Article bookmarked successfully', addBookmark);
+  }
+
+  /**
+   * @description Remove Article from Bookmark
+   * @param {Object} req - request object
+   * @param {Object} res - response object
+   * @returns {Object} Success Message
+   */
+  static async unbookmarkArticle(req, res) {
+    const userId = req.user.id;
+    const { articleId } = req.params;
+    const article = await models.Article.findByPk(articleId);
+    if (!article) return utils.errorStat(res, 404, 'Article not found');
+    const bookmarked = await models.Bookmark.findOne({
+      where: {
+        [Op.or]: [{ articleId }]
```

Comment:
You're providing one value in the array. Why do you need the [Op.or]?

Reply:
When I specified jusst the articleid without an option while using the findOne method, sequelize threw me an error that I needed an option, I did some research and turns out I needed the [Op.or] (for the findOne method